pkg/cmd/cli/config/store.go

```diff
@@ -0,0 +1,185 @@
+// to persist configs by being explicit about the file to save.
+type ConfigStore struct {
+	Config         *clientcmdapi.Config
+	Path           string
+	providerEngine string
+}
+
+type ConfigStoreNotFoundError struct {
+	msg string
+}
+
+func (e *ConfigStoreNotFoundError) Error() string {
+	return e.msg
+}
+
+func (c *ConfigStore) FromOpenShift() bool {
+	return c.providerEngine == fromOpenShift
+}
+
+func (c *ConfigStore) FromKube() bool {
+	return c.providerEngine == fromKube
+}
+
+// Load a ConfigStore, first trying the explicit path to a config file provided as argument,
+// and if it was not provided, load using the priority conventions used by the
+// ClientConfigLoadingRules. Error if none can be found.
+func Load(clientCfg *client.Config, explicitPath string) (*ConfigStore, error) {
+	// if an explicit path were provided only tries it
+	path := explicitPath
+	if len(path) > 0 {
+		config, err := tryToLoad(path, fromOpenShift)
+		if err == nil {
+			return config, nil
+		} else {
+			return nil, err
+		}
+	}
+
+	for _, path := range OpenShiftClientConfigFilePriority() {
+		config, err := tryToLoad(path, fromOpenShift)
+		if err == nil {
+			return config, nil
+		}
+	}
+
+	for _, path := range KubeClientConfigFilePriority() {
+		config, err := tryToLoad(path, fromKube)
+		if err == nil {
+			return config, nil
+		}
+	}
+
+	err := &ConfigStoreNotFoundError{"Config file not found in any of the expected locations."}
+	glog.V(4).Infof(err.Error())
+
+	return nil, err
+}
+
+// Create a new file to store configs and returns the ConfigStore that represents it.
+func CreateEmpty() (*ConfigStore, error) {
+	configPathToCreateIfNotFound := fmt.Sprintf("%v/%v", os.Getenv("HOME"), OpenShiftConfigHomeDirFileName)
+
+	glog.V(3).Infof("A new config will be created at: %v ", configPathToCreateIfNotFound)
+
+	newConfig := clientcmdapi.NewConfig()
+
+	if err := os.MkdirAll(fmt.Sprintf("%v/%v", os.Getenv("HOME"), OpenShiftConfigHomeDir), 0755); err != nil {
+		return nil, fmt.Errorf("Tried to create a new config file but failed while creating directory %v: %v", OpenShiftConfigHomeDirFileName, err)
+	}
+	glog.V(5).Infof("Created directory %v", "~/"+OpenShiftConfigHomeDir)
+
+	if err := clientcmd.WriteToFile(*newConfig, configPathToCreateIfNotFound); err != nil {
+		return nil, fmt.Errorf("Tried to create a new config file but failed with: %v", err)
+	}
+	glog.V(5).Infof("Created file %v", configPathToCreateIfNotFound)
+
+	config, err := tryToLoad(configPathToCreateIfNotFound, fromOpenShift)
+	if err != nil {
+		return nil, err
+	}
+
+	return config, nil
+}
+
+// Same as Load, but creates a file if none can be found.
+func LoadOrCreateEmpty(clientCfg *client.Config, explicitPath string) (*ConfigStore, bool, error) {
+	created := false
+
+	configStore, err := Load(clientCfg, explicitPath)
+
+	if err != nil {
+		if _, isNotFound := err.(*ConfigStoreNotFoundError); isNotFound {
+			configStore, err = CreateEmpty()
```

Comment:
I considered it but I'm not completely sure it is desired. It seems like if you provide a path in a `--config` flag you are expecting to use something you already have, otherwise you don't know about it and just want `osc` to take care of it.
You don't even have tab autocomplete to provide the path to a file you don't have.  :)
@liggitt ?